### PR TITLE
fix_for_Plugin_Toolkit_check_ckan_version #7777

### PR DESCRIPTION
**Pull request by Vishal-NEC**
#Fix for issue #7777

### Features:

- [X] includes tests covering changes
- [X] includes updated documentation
- [X] includes user-visible changes
- [X] includes API changes
- [X] includes bugfix for possible backport

### CODE REVIEW

File: ckan/plugins/toolkit.py [hunk 1]
```diff
@@ -220,17 +212,19 @@ def check_ckan_version(
...
-    if max_version:
-        max_required = _version_str_2_list(max_version)
-        if current > max_required:
-            return False
-    return True
+    current = pv.parse(ckan.__version__)
+    try:
+        at_least_min = (min_version is None) or (current >= pv.parse(min_version))
+    except pv.InvalidVersion:
+        # TODO: This should be a custom error per the CKAN dev guidelines
```
Comment (smotornyuk) [inline ckan/plugins/toolkit.py:219]: Please, finalize the PR. Instead of adding new TODO's into codebase, implement whatever you suggesting here

**Comment by Vishal-NEC** [comment]
Hi @smotornyuk  I have Update the PR as per your Review. I think no need to implement this TODO in Toolkit.py as already a exception is raised.

Please let me know if anything I am missing.

**Comment by smotornyuk** [comment]
You need to fix errors reported by flake8. Looks like original solution from #7777 was not formatted properly

**Comment by Vishal-NEC** [comment]
Yes, There are Flake8 mistake. Let me do it correct.

Thanks for confirming. @smotornyuk

**Pull request merged**